src/board.cpp

```diff
@@ -0,0 +1,190 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include "board.h"
+#include "cpu.h"
+
+
+Board::Board(GUI *gui, int board_size) : gui_(gui), board_size_(board_size) {
```

Comment:
i see you're initializing to nullptr. If you _really_ want to keep that, then you can make a `std::unique_ptr` i guess... potentially use `std::optional`? Think about it. If you decide to keep the ptr then that's okay

Reply:
1) We made it a class of static functions. This means I need to pass it in in some way now right?

3) Ree, I need to recall which ptr to use... :)

Reply:
You shouldn't need to create an object if it's only static functions.
Instead of
```cpp
GUI g;
g.print();
```
It would be
```cpp
GUI::print();
```

Reply:
So back to the OOP vs iterative ideology, should GUI just have been a namespace instead? Cause now that both function the same way?

Reply:
Nah I'd keep it in a class. Better organization

Reply:
In basic C++ you should have everything in a class except for like specific helper functions (like that rtrim thing or whatever it was called0